praw/models/reddit/subreddit.py

```diff
@@ -873,6 +877,120 @@ def submit(
+        :param flair_id: The flair template to select (default: None).
+        :param flair_text: If the template's ``flair_text_editable`` value is
+            True, this value will set a custom text (default: None).
+        :param resubmit: When False, an error will occur if the URL has already
+            been submitted (default: True).
+        :param send_replies: When True, messages will be sent to the submission
+            author when comments are made to the submission (default: True).
+        :param nsfw: Whether or not the submission should be marked NSFW
+            (default: False).
+        :param spoiler: Whether or not the submission should be marked as
+            a spoiler (default: False).
+        :param discussion_type: Set to ``CHAT`` to enable live discussion
+            instead of traditional comments (default: None).
+        :returns: A :class:`.Submission` object for the newly created submission.
+
+        If ``image_path`` in ``images`` refers to a file that is not an image, PRAW will
+        raise a :class:`.ClientException`.
+
+        For example to submit an image gallery to ``r/reddit_api_test`` do:
+
+        .. code-block:: python
+
+            title = "My favorite pictures"
+            image = "/path/to/image.png"
+            image2 = "/path/to/image2.png"
+            image3 = "/path/to/image3.png"
+            images = [
+               {
+                   "image_path": image
+               },
+               {
+                   "image_path": image2,
+                   "caption": "Image caption 2",
+               },
+               {
+                   "image_path": image3,
+                   "caption": "Image caption 3",
+                   "outbound_url": "https://example.com/link3"
+               }
+            ]
+            reddit.subreddit("reddit_api_test").submit_gallery(title, images)
+
+        .. seealso ::
+
+           * :meth:`.submit` to submit url posts and selftexts
+           * :meth:`.submit_image`. to submit single images
+           * :meth:`.submit_video`. to submit videos and videogifs
+
+        """
+        data = {
+            "sr": str(self),
+            "resubmit": bool(resubmit),
+            "api_type": "json",
+            "show_error_list": True,
+            "title": title,
+            "sendreplies": bool(send_replies),
+            "nsfw": bool(nsfw),
+            "spoiler": bool(spoiler),
+            "validate_on_submit": self._reddit.validate_on_submit,
+            "items": [],
+        }
+        for key, value in (
+            ("flair_id", flair_id),
+            ("flair_text", flair_text),
+            ("collection_id", collection_id),
+            ("discussion_type", discussion_type),
+        ):
+            if value is not None:
+                data[key] = value
+        if not len(images) > 1:
+            raise TypeError("At least 2 images is required.")
+        for image in images:
+            if "image_path" not in image:
```

Comment:
I would make this check a bit stronger at this point and verify that it is actually a file path, otherwise, we don't get the client-side exception until we start uploading the images.

    if not os.path.isfile(image.get("image_path", ""))

Or something similar.